src/NHibernate/Impl/SessionImpl.cs

```diff
@@ -2206,36 +2124,21 @@ public ISession SetBatchSize(int batchSize)
 			return this;
 		}
 
-
 		public ISessionImplementor GetSessionImplementation()
 		{
 			return this;
 		}
 
-		public ISession GetChildSession()
-		{
-			using (new SessionIdLoggingContext(SessionId))
-			{
-				if (rootSession != null)
-				{
-					return rootSession.GetChildSession();
-				}
-
-				CheckAndUpdateSessionStatus();
-
-				if (_childSession == null)
-				{
-					log.Debug("Creating child session.");
-					_childSession = new SessionImpl(this);
-				}
-
-				return _childSession;
-			}
-		}
-
-		public override IInterceptor Interceptor
+		// Obsolete since v5
+		[Obsolete("Please use SessionWithOptions instead.")]
+		public ISession GetSession(EntityMode entityMode)
 		{
-			get { return interceptor; }
+			return SessionWithOptions()
+				.Connection()
+				.ConnectionReleaseMode()
+				.FlushMode()
+				.Interceptor()
+				.OpenSession();
```

Comment:
Have to do all that with the new method if you want to be as close as possible... But we have no covering tests for that indeed: in the tests, I have just call `Connection()`, and that is enough for all of them.